core/Pos/Core/Timestamp.hs

```diff
@@ -45,3 +47,9 @@ timestampF = build
 -- Get the current time as a timestamp
 getCurrentTimestamp :: IO Timestamp
 getCurrentTimestamp = Timestamp . round . (*1000000) <$> getPOSIXTime
+
+diffTimestamp :: Timestamp -> Timestamp -> Microsecond
+diffTimestamp t1 t2 = getTimestamp t1 - getTimestamp t2
+
+addMicrosecondsToTimestamp :: Microsecond -> Timestamp -> Timestamp
```

Comment:
Usually it's shorter to write `Timestamp ms + b` than `addMicrosecondsToTimestamp ms b` (which are same things). So I am not sure this function is needed.
If you think it's needed, I don't mind having it (it's subjective and I don't want insist on subjective changes).